cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -441,12 +455,16 @@ func (csr *ClusterStateRegistry) updateNodeGroupMetrics() {
 	metrics.UpdateNodeGroupsCount(autoscaled, autoprovisioned)
 }
 
-// IsNodeGroupSafeToScaleUp returns true if node group can be scaled up now.
-func (csr *ClusterStateRegistry) IsNodeGroupSafeToScaleUp(nodeGroup cloudprovider.NodeGroup, now time.Time) bool {
+// IsNodeGroupSafeToScaleUp returns information about node group safety to be scaled up now.
+func (csr *ClusterStateRegistry) IsNodeGroupSafeToScaleUp(nodeGroup cloudprovider.NodeGroup, now time.Time) NodeGroupScalingSafety {
 	if !csr.IsNodeGroupHealthy(nodeGroup.Id()) {
-		return false
+		return NodeGroupScalingSafety{SafeToScale: false, Healthy: false, BackoffStatus: nil}
 	}
-	return !csr.backoff.IsBackedOff(nodeGroup, csr.nodeInfosForGroups[nodeGroup.Id()], now)
+	backoffStatus := csr.backoff.BackoffStatus(nodeGroup, csr.nodeInfosForGroups[nodeGroup.Id()], now)
```

Comment:
Hmm, wouldn't it be simpler to have BackoffStatus as a struct and not a pointer, and just always set it?  I'm slightly worried about somebody assuming it's always set and hitting a panic.